vault/activity_log.go

```diff
@@ -975,7 +973,7 @@ func (a *ActivityLog) SetConfig(ctx context.Context, config activityConfig) {
 	if !a.enabled && a.currentSegment.startTimestamp != 0 {
 		a.logger.Trace("deleting current segment")
 		a.deleteDone = make(chan struct{})
-		go a.deleteLogWorker(a.currentSegment.startTimestamp, a.deleteDone)
+		go a.deleteLogWorker(ctx, a.currentSegment.startTimestamp, a.deleteDone)
```

Comment:
Since SetConfig is invoked from a request context, it should pass b.Core.activeContext to any workers it spawns.

Reply:
1b5ca5c8d